public/app/plugins/datasource/influxdb/components/RawInfluxQLEditor.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FC, useState, useEffect } from 'react';
+import { TextArea, InlineFormLabel, Input, Select, HorizontalGroup } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { ResultFormat, InfluxQuery } from '../types';
+
+const RESULT_FORMATS: Array<SelectableValue<ResultFormat>> = [
+  { label: 'Time series', value: 'time_series' },
+  { label: 'Table', value: 'table' },
+  { label: 'Logs', value: 'logs' },
+];
+
+const DEFAULT_RESULT_FORMAT: ResultFormat = 'time_series';
+
+type Props = {
+  query: InfluxQuery;
+  onQueryChange: (query: InfluxQuery) => void;
+  onRunQuery: () => void;
+};
+
+// we handle 3 fields: "query", "alias", "resultFormat"
+// "resultFormat" changes are applied immediately
+// "query" and "alias" changes only happen on onblur
```

Comment:
Can we put this into a component test? Should be easy to test with react-testing-library we use now. Or is that handled in the editor tests? I don't see it there.